x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/mapper/VectorEncoderDecoder.java

```diff
@@ -31,7 +32,39 @@ public static int denseVectorLength(Version indexVersion, BytesRef vectorBR) {
      */
     public static float decodeVectorMagnitude(Version indexVersion, BytesRef vectorBR) {
         assert indexVersion.onOrAfter(Version.V_7_5_0);
+        int offset = vectorBR.offset + vectorBR.length - INT_BYTES;
+        int intValue = ((vectorBR.bytes[offset] & 0xFF) << 24)   |
+            ((vectorBR.bytes[offset+1] & 0xFF) << 16) |
+            ((vectorBR.bytes[offset+2] & 0xFF) <<  8) |
+            (vectorBR.bytes[offset+3] & 0xFF);
+        return Float.intBitsToFloat(intValue);
+    }
```

Comment:
I removed using ByteBuffer while retrieving a vector's magnitude. First the benchmarks shows that the performance is almost the same, second we don't need to create extra ByteBuffer objects that would need be garbage collected later.

10 dims

```
Benchmark                                    Mode  Cnt  Score   Error  Units
VectorFunctionBenchmark.decodeMag            avgt   30  3.953 ± 0.101  ns/op
VectorFunctionBenchmark.decodeMagWithBuffer  avgt   30  3.504 ± 0.053  ns/op
```

100 dims
```
VectorFunctionBenchmark.decodeMag            avgt   30  3.728 ± 0.103  ns/op
VectorFunctionBenchmark.decodeMagWithBuffer  avgt   30  3.276 ± 0.067  ns/op
```

512 dims
```
VectorFunctionBenchmark.decodeMag            avgt   30  3.752 ± 0.099  ns/op
VectorFunctionBenchmark.decodeMagWithBuffer  avgt   30  3.377 ± 0.023  ns/op
```

Reply:
@jtibshirani Thanks for the feedback. I agree to keep the PR focused, and I removed this change from this PR. But  may be we can follow it up later.    I wonder what is the implication of creating and garbage collecting of a `ByteBuffer` object, is it work extra `0.5ns`?